webxr/render_state_update.https.html

```diff
@@ -0,0 +1,88 @@
+<!DOCTYPE html>
+<script src="/resources/testharness.js"></script>
+<script src="/resources/testharnessreport.js"></script>
+<script src="resources/webxr_util.js"></script>
+<script src="resources/webxr_test_constants.js"></script>
+<canvas />
+
+<script>
+let fakeDeviceInitParams = TRACKED_IMMERSIVE_DEVICE;
+
+let testSessionEnded = function(session, fakeDeviceController, t) {
+  return new Promise((resolve, reject) => {
+    resolve(session.end().then(() => {
+      assert_throws_dom('InvalidStateError', () => session.updateRenderState({}));
+    }));
+  });
+};
+
+
+let testBaseLayer = function(session, fakeDeviceController, t, sessionObjects) {
+  return new Promise((resolve, reject) => {
+    navigator.xr.test.simulateUserActivation(() => {
+      navigator.xr.requestSession('inline').then((tempSession) => {
+        assert_not_equals(session, tempSession);
+        assert_throws_dom('InvalidStateError', () => session.updateRenderState({ baseLayer : new XRWebGLLayer(tempSession, sessionObjects.gl), }));
+      });
+      resolve();
+    });
+  });
+};
+
+let testFieldOfView = function(session, fakeDeviceController, t) {
+  return new Promise((resolve, reject) => {
+    assert_throws_dom('InvalidStateError', () => session.updateRenderState({ inlineVerticalFieldOfView : Math.PI, }));
```

Comment:
This test already exists via: webxr/render_state_vertical_fov_inline.https.html

Reply:
Ah, and also throwing isn't technically compliant: https://immersive-web.github.io/webxr/#apply-the-pending-render-state

Reply:
I think they're testing two different things. I'm trying to test the updateRenderState() API (which indeed throws) while the test you mention is actually testing the apply render state if I am not wrong

Reply:
I disagree, step 11 from that algorithm https://immersive-web.github.io/webxr/#dom-xrsession-updaterenderstate
Doesn't mention any validation to inlineVerticalFieldOfView except for throwing if it's set and being applied to an immersive session.
Applying the pending render state is where any validation actually happens, and that happens by clamping.

Reply:
Actually... I just looked closer, and this *is* setting to an immersive session (Not sure why I assumed it was setting on an inline session...).

That is also already tested via: webxr/render_state_vertical_fov_immersive.https.html

Reply:
> I disagree, step 11 from that algorithm https://immersive-web.github.io/webxr/#dom-xrsession-updaterenderstate
> Doesn't mention any validation to inlineVerticalFieldOfView except for throwing if it's set and being applied to an immersive session.
> Applying the pending render state is where any validation actually happens, and that happens by clamping.

Then I might be missunderstanding the specs. What I am trying to test is step 4 of `updateRenderState()`, that's before the step 11 you mention.

Reply:
Just to be crystal clear, I don't want to test the validation at all (as you said there are 2 other tests for that), I just want to test the first 5 steps of the update algorithm.

Reply:
Yeah; sorry, my initial understanding was wrong and I thought it was the step 11; however step 4 *is* still tested (though it doesn't use assert_throws and ~implements that itself) in this test: https://github.com/web-platform-tests/wpt/blob/master/webxr/render_state_vertical_fov_immersive.https.html
(Similar name to the one I initially linked, but replicates this test).

Reply:
OK, I understand that you prefer to remove it then. I'll understand silence as a confirmation :)